adam-core/src/main/scala/org/bdgenomics/adam/rdd/settheory/SetTheory.scala

```diff
@@ -0,0 +1,346 @@
+package org.bdgenomics.adam.rdd.settheory
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.ReferenceRegion
+import scala.collection.mutable.ListBuffer
+
+private[settheory] sealed abstract class SetTheory extends Serializable {
+
+  protected val threshold: Long
+  protected val optPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]]
+
+  /**
+   * The condition that should be met in order for the primitive to be
```

Comment:
"SetTheory" doesn't work for me as a class and package name.
Perhaps for package `rdd.sets`, and `SetOperation` for the top level abstract class?